woocommerce/wc-api-dev
# Add hotfix that prevents the cheque payment method from being auto-enabled.

See https://github.com/Automattic/wp-calypso/issues/16658 and https://github.com/Automattic/wp-calypso/pull/16663 for some history.

By default WC auto-enables the cheque payment method. We don't want this auto-enabled for Store on WP.com. Luckily, we are provided a filter where we can pass defaults, which are used if settings are not stored for the site (i.e. user hasn't configured the payment method). It also cleans up the default text.

It also adds a constant for easily disabling all these WP.com specific hotfixes.

To Test:
* Install this branch/plugin over your existing `wc-api-dev` copy.
* Unless you just set-up a fresh store, you will need to remove your settings for the cheque payment method. What I did for my AT site, was edit `wc-api-dev.php`'s `init` using the `wp-admin` plugin editor and added a `delete_option( 'woocommerce_cheque_settings' );`. I did a page load, and then removed this line.
* View the checkout page and make sure the `cheque` payment method is not displayed.
* Go to `http://calypso.localhost:3000/store/settings/:site` and enable the `cheque` payment method. Under setup, you should see the defaults.
* View the checkout page and make sure the `cheque` payment method is displayed.
* Run `phpunit` and make sure all tests pass (just to be safe).
